### ISSUE TITLE
Feature Request: Rust Support

### ISSUE THREAD

**Issue by ghost** [bazelbuild/intellij#675]
I've been working on integrating a Rust project with a large multi-language repository, and we're successfully building Rust binaries via `rules_rust` and `cargo raze`. However, we have not been able to get the projects working within CLion or Intellij IDEA, I think in part because the plugin does not yet support it.

Is Rust support something that the Intellij plugin authors would be open to? I am happy to dedicate time to making this come to fruition, if it is desired. Roughly speaking, what would it take to get this working?

**Comment by petroseskinder** [comment]
@jin there seems to already be support for go, cpp, dart, javascript, python, scala, kotlin and java. 

What is the process for adding a new language? Barring scala which wix added support for, each of the existing languages are used internally within Google. I presume one concern is who would be on the hook for supporting the language support after the initial implementation

**Comment by jin** [comment]
Currently, we don't have bandwidth or domain expertise to support new languages like Rust, and there isn't a process to add a new language yet. It's possible to work on a fork of this project to add Rust support, but until the details are ironed out, we cannot support this yet.

One of the things that's high on my list to do is to implement a pluggable framework so that rule owners / domain experts (e.g. rules_rust) can integrate with this plugin without having to extend the core IntelliJ plugin, similar to how Starlark has provided Bazel core with extensibility.

**Comment by mattboardman** [comment]
Any updates on this? Will this be scoped out or prioritized in/for 2021?

**Comment by vlad20012** [comment]
By the way, could someone link an example of a project that uses Bazel with Rust?

**Comment by jelmansouri** [comment]
https://github.com/bazelbuild/rules_rust/tree/master/examples has a couple of samples.
I can build a repo that mimics the common components used in internal monorepos using bazel if it might help.
And here you can find an example of crates where cargo raze was used to generate build files:
https://github.com/bazelbuild/rules_rust/tree/master/bindgen/raze/remote

**Comment by briansmith** [comment]
> Currently, we don't have bandwidth or domain expertise to support new languages like Rust, and there isn't a process to add a new language yet. It's possible to work on a fork of this project to add Rust support, but until the details are ironed out, we cannot support this yet.

Is forking this project still seen as the best way to getting Rust support added given the current state of the project?

Would you accept a PR to add direct Rust support now? Or...

> One of the things that's high on my list to do is to implement a pluggable framework so that rule owners / domain experts (e.g. rules_rust) can integrate with this plugin without having to extend the core IntelliJ plugin, similar to how Starlark has provided Bazel core with extensibility.

...is this completely blocked on implementing such a pluggable framework?

**Comment by alexjpwalker** [comment]
I've opened a PR which integrates with the IntelliJ Rust plugin to deliver Rust support.

- #3222 

It depends on https://github.com/intellij-rust/intellij-rust/pull/8491. I've uploaded an example project, and demo of the plugin's capabilities, at https://github.com/vaticle/bazel-intellij-rust-example

https://user-images.githubusercontent.com/14253102/151391801-a7551fcb-0322-4ed5-adc1-f11784a3b106.mov

The PR's architecture is suboptimal, looking for some insider knowledge to bring it to production grade.
